.github/workflows/auto-generated-checker.yml

```diff
@@ -0,0 +1,40 @@
+name: Auto-Generated Code Checker
+on:
+  push:
+    tags:
+      - "*"
+    branches:
+      - master
+      - dev
+  pull_request:
+
+jobs:
+  protobuf_codegen:
+    name: Protobuf
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.19.12'
+          check-latest: true
+      - uses: bufbuild/buf-setup-action@v1.23.1
+      - shell: bash
+        run: scripts/protobuf_codegen.sh
+      - shell: bash
+        run: .github/workflows/check-clean-branch.sh
+  mock_gen:
+    name: Mocks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.19.12'
+          check-latest: true
+      - shell: bash
+        run: scripts/mock.gen.sh
+      - shell: bash
+        run: rm -rf gomock_reflect*
```

Comment:
Is this mostly for if the previous run failed in a weird state?

Reply:
Yeah- thinking about it again we should fail if that file still exists. Updated the PR